tests/test_mattermost.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import logging
+from PaperBee.papers.mattermost_papers_formatter import MattermostPaperPublisher
+from tests.sample_papers import sample_papers
+from unittest.mock import MagicMock
+import yaml
+import os
+
+@pytest.fixture
+def papers():
+    return sample_papers
+
+@pytest.fixture
+def config():
+    config_path = os.environ.get("PAPERBEE_CONFIG", "files/config_template.yml")
+    with open(config_path) as f:
+        return yaml.safe_load(f)
+
+def test_format_papers(papers):
+    papers_out, preprints_out = MattermostPaperPublisher.format_papers(papers)
+    assert isinstance(papers_out, list)
+    assert isinstance(preprints_out, list)
+    assert len(preprints_out) == len(papers)
+    assert all("[" in p and "](" in p for p in preprints_out)
+
+def test_build_message(papers):
+    publisher = MattermostPaperPublisher(
+        logger=logging.getLogger(__name__),
+        url="dummy", token="dummy", team="dummy", channel="dummy", driver=MagicMock()
```

Comment:
Don't use mocks and dummy IDs, we need a real test. If you don't have mattermost, read these variables from config, we will run it